tests/grid/test_constructors.py

```diff
@@ -177,27 +177,85 @@ def test_network_from_file():
     assert mg.xy_of_reference == (12345, 678910)
 
 
+def test_framed_voronoi_edge_nodes():
+    grid = FramedVoronoiGrid(
+        (5, 6), xy_spacing=(10, 15), xy_of_lower_left=(1, 2), xy_min_spacing=7.5
+    )
+
+    expected_x = (
+        np.array(
+            [
+                [0, 10, 20, 30, 40, 50],
+                [0, 99, 99, 99, 99, 50],
+                [0, 99, 99, 99, 99, 50],
+                [0, 99, 99, 99, 99, 50],
+                [0, 10, 20, 30, 40, 50],
+            ]
+        )
+        + 1.0
+    )
+    expected_y = (
+        np.array(
+            [
+                [0, 0, 0, 0, 0, 0],
+                [15 - 7.5 / 2 - 0.001, 99, 99, 99, 99, 15 + 7.5 / 2 + 0.001],
+                [30 - 7.5 / 2 - 0.001, 99, 99, 99, 99, 30 + 7.5 / 2 + 0.001],
+                [45 - 7.5 / 2 - 0.001, 99, 99, 99, 99, 45 + 7.5 / 2 + 0.001],
+                [60, 60, 60, 60, 60, 60],
+            ]
+        )
+        + 2.0
+    )
+    actual_x = grid.x_of_node.reshape(grid.shape)
+    actual_y = grid.y_of_node.reshape(grid.shape)
+
+    assert_array_almost_equal(expected_x[0, :], actual_x[0, :])
+    assert_array_almost_equal(expected_x[-1, :], actual_x[-1, :])
+    assert_array_almost_equal(expected_x[:, 0], actual_x[:, 0])
+    assert_array_almost_equal(expected_x[:, -1], actual_x[:, -1])
+
+    assert_array_almost_equal(expected_y[0, :], actual_y[0, :])
+    assert_array_almost_equal(expected_y[-1, :], actual_y[-1, :])
+    assert_array_almost_equal(expected_y[:, 0], actual_y[:, 0])
+    assert_array_almost_equal(expected_y[:, -1], actual_y[:, -1])
+
+
+@pytest.mark.parametrize("xy_min_spacing", (1, 0, 3, (3, 4), (1, 10)))
+def test_framed_voronoi_min_spacing(xy_min_spacing):
+    grid = FramedVoronoiGrid(
+        (100, 200), xy_spacing=(3, 10), xy_min_spacing=xy_min_spacing
+    )
+
+    assert np.all(grid.length_of_link >= np.min(xy_min_spacing))
+
+
+@pytest.mark.parametrize("xy_min_spacing", (10, (3, 10), (4, 11)))
```

Comment:
I *think* all of these min spacings should result in an error.

Reply:
Right